### PR TITLE
Setup Android app MVP with skeleton Welcome screen

### PR DESCRIPTION
**Pull request by crocsandcoffee**
## 📑 What does this PR do?

- Sets up the initial android app with a no-op Welcome Screen
- Minor fixes to different grade scripts to build the android app
- Fixes build issues with desktop related dependencies being used in common sources. For now I moved the affected files to a `desktopMain` source but we will want to fix move back shared composables and utility without the offending desktop dependencies

# ✅ Checklist

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [x] I have updated the documentation as required
- [x] All the tests have passed

## 🧪 How can this PR been tested?
- Checkout this branch, and run the gradle command in the README which will build and install the APK on your connected emulator or device

## 🧾 Tasks Remaining: (List of tasks remaining to be implemented)

- Replace the placeholder UI with a polished welcome screen
- Hook up navigation graph for Login and Sign Up Screens
- Probably need to look at updating the Mastodon Theme so we can start using that throughout all composables.

## 🖼️ Screenshots (if applicable):

Reminder - this screen is just a dummy UI. I just wanted to get everything up and running and commit the android setup to the repo

### CODE REVIEW

**Review by roshanrai06 — APPROVED** [review]

File: build.gradle.kts
Comment (crocsandcoffee) [inline build.gradle.kts:4]: Removed - this is not needed.

File: build.gradle.kts
Comment (crocsandcoffee) [inline build.gradle.kts:5]: For now, I am placing versions via extras on the project. We can decide how/where we want to handle the versioning of dependencies across the entire project to ensure version parity

File: ui/welcome/src/commonMain/kotlin/social/androiddev/welcome/composables/WelcomeScreen.kt
Comment (crocsandcoffee) [inline ui/welcome/src/commonMain/kotlin/social/androiddev/welcome/composables/WelcomeScreen.kt:20]: 1) We should start using the `MastodonTheme` properly across all UI. This will open the door for someone else to tackle supporting Material You and building upon the theme as we focus on getting MVP functionality done...

2) Ignore this layout, it's just a dummy placeholder added to see if everything worked end-to-end. In a follow-up PR we can actually implement the full Welcome Screen with LogIn/SignUp flows.

File: build.gradle.kts
Comment (roshanrai06) [reply build.gradle.kts:5]: do we want to keep min SDK to 23? it could be higher since API 24 or 26 covers more than 80%.

Comment (crocsandcoffee) [reply build.gradle.kts:5]: Not sure.. I haven't thought that far. We can chat over it in slack and see what the group thinks. I don't feel strongly here. But, If we do want to increase adoption and availability, especially globally, then probably a minSDK of 6 is the way to go

File: build.gradle.kts
Comment (thebino) [inline build.gradle.kts:16]: I would not raise the versions in this PR. I've had some issues when updating, so stuck with these older versions.

**Review by thebino — CHANGES_REQUESTED** [review]
Do not move the 
`ui/common/src/commonMain/kotlin/social/androiddev/common/composables/`
files into `desktopMain/`
They should be reusable from all platforms

File: ui/welcome/src/androidMain/AndroidManifest.xml
Comment (thebino) [inline ui/welcome/src/androidMain/AndroidManifest.xml:3]: should be `social.androiddev.ui.welcome`

File: ui/welcome/src/commonMain/kotlin/social/androiddev/welcome/composables/WelcomeScreen.kt
Comment (thebino) [reply ui/welcome/src/commonMain/kotlin/social/androiddev/welcome/composables/WelcomeScreen.kt:20]: Let's move the `MastodonTheme` into the highest layer so the composable and screens are free to use in a separate app.

**Comment by crocsandcoffee** [comment]
> Do not move the 
> `ui/common/src/commonMain/kotlin/social/androiddev/common/composables/`
> files into `desktopMain/`
> They should be reusable from all platforms

Let's chat offline. They are referencing desktop related dependencies and as such cause build issues with the android app.

I just moved them for the quick fix to unblock getting the android app working, but you'll need to rework those utils and composables to pull out the desktop related code outside of the common/shared source set.

File: ui/welcome/src/commonMain/kotlin/social/androiddev/welcome/composables/WelcomeScreen.kt
Comment (crocsandcoffee) [reply ui/welcome/src/commonMain/kotlin/social/androiddev/welcome/composables/WelcomeScreen.kt:20]: Yup - take note of my comment. I just threw everything here to see it working end-to-end, including the theme. I'll be pulling that out and placing it at root in the next PR when I actually implement this screen.

File: build.gradle.kts
Comment (crocsandcoffee) [reply build.gradle.kts:16]: This fixes warnings and (I think) an error I was getting initially with the android setup.

I'm guessing you probably copied over the templates for KMP which had outdated versions.. i tested running both desktop and the android app with this PR and these changes.

File: ui/welcome/src/androidMain/AndroidManifest.xml
Comment (racka98) [reply ui/welcome/src/androidMain/AndroidManifest.xml:3]: These should all be moved to gradle file (namespace). They will no longer supported in future feature of AGP. AGP 7.3.x already throws a bunch of warnings about this

File: build.gradle.kts
Comment (racka98) [reply build.gradle.kts:5]: minSDK 23 will cause a lot of headache with desugaring when KMP and kotlinx-datetime are involved.

Comment (thebino) [reply build.gradle.kts:5]: Support as low as possible but try to reduce the maintenance only to cover legacy versions.
Always keep in mind, there are regions on earth, where not the latest phone is in everyone’s pocket.

I’m of course fine with raising to 24 or 26 as well. At least when we add video playback to the timeline, we’ll need to raise it most probably.

Comment (racka98) [reply build.gradle.kts:5]: Also, I think using Versions catalog will be easier. 
I can make a PR adding them once this is merged

File: ui/welcome/src/androidMain/AndroidManifest.xml
Comment (crocsandcoffee) [reply ui/welcome/src/androidMain/AndroidManifest.xml:3]: @racka98 mind doing that in your follow-up PR with the version catalog?

**Pull request merged**
